### DISCUSSION TITLE
Converting old printer driver to IPP Everywhere?

### DISCUSSION THREAD

**Discussion by plvlnisse** [OpenPrinting/libcups#88] [Q&A] [answered]
Hi,

I have a Brother HL-4150CDN that I intend to use for many more years (touch wood) as it has at least 90000 pages worth of lifetime left :smile: 

Forgot to state in OP that the printer supports ipp and ipps as well as identifies itself fine with CUPS. Its just the filtering/setup/interfacing part that needs to be updated to ippevprinter.

I have access to software from the manufacturer (openly available) that clearly shows how the current filter/PPD based interaction is handled. So I guess this is a perfect little project to implement a IPP/Everywhere-ish interface using ippeveprinter and I intend to do so and share.

I can not see any licensing issues (so far at least) wit this approach and chances that the manufacturer will do this are slim, I think.

If anyone here has pointers to documentation on how this can be made, preferably in a way that is easily incorporated into distributions of Cups for users and package maintainers, it would be much appreciated.

I will of course RTFM my self and continue to search the interweb, but if there are lessons to learn and I can skip their trial-and-error phase I don't see that as cheating.

Thanks in advance!

**Comment by michaelrsweet** [comment, marked as answer]
> Response copied from issue #89.

So generally the best advice is to develop a printer application using [PAPPL](https://www.msweet.org/pappl/). But Till Kamppeter has done good things with the [pappl-retrofit](https://github.com/OpenPrinting/pappl-retrofit) printer application, which will run the existing driver as-is. A "native" printer application will provide the best capabilities (and you have the opportunity to report more printer status/configuration info if supported by the printer) but if you are happy with the current functionality the retrofit approach is quick and easy.
